src/preamble.js

```diff
@@ -2509,7 +2509,7 @@ function integrateWasmJS(Module) {
       } else if (curr === 'interpret-asm2wasm' || curr === 'interpret-s-expr' || curr === 'interpret-binary') {
         if (exports = doWasmPolyfill(global, env, providedBuffer, curr)) break;
       } else {
-        throw 'bad method: ' + curr;
+        Module['quit'](1, 'bad method: ' + curr);
```

Comment:
Got it, thanks; just took care of this in a force push over the original commit.